compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -2026,7 +2026,10 @@ private BType computeIntersectionType(BLangIntersectionTypeNode intersectionType
                                           constituentBTypes, existingErrorDetailType, env);
         }
 
-        if (types.isInherentlyImmutableType(potentialIntersectionType)) {
+        if (types.isInherentlyImmutableType(potentialIntersectionType) ||
```

Comment:
Shouldn't enum anyway be added in isInherentlyImmutableType()?

Reply:
That's what I though initially too. But so far we've had kind of a direct match in this check with the spec https://ballerina.io/ballerina-spec/spec.html#readonly-type-descriptor.

Also, with explicit unions (vs enums) also the inherently immutable check would return false.

e.g.,
```ballerina
public const RED = "RED";
public const GREEN = "GREEN";
public const BLUE = "BLUE";
public type Color RED|GREEN|BLUE;
```

So for consistently, I think we should continue this separation. WDYT?

Reply:
Cool. Makes sense then